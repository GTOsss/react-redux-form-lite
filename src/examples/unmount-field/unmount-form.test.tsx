import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import UnmountForm from './unmount-form';
import {reducer} from '../../index';

const renderComponent = () => {
  const reduxThunkTester = new ReduxThunkTester();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}><UnmountForm /></Provider>,
  );

  return {reduxThunkTester, store, component};
};

describe('<UnmountField />', () => {
  test('Render UnmountField (inputs)', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render UnmountField: inputs', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render UnmountField: store', () => {
    const {store} = renderComponent();
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });

  test('Render UnmountField: inputs after change', () => {
    const {component} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    expect(component.find('input')).toMatchSnapshot();
  });

  test('UnmountField: mount field (action history)', () => {
    const {component, reduxThunkTester} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    expect(reduxThunkTester.getActionHistoryStringify()).toMatchSnapshot();
  });

  test('UnmountField: mount field (store)', () => {
    const {component, store} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });

  test('UnmountField: render with field and then remove it (action history)', () => {
    const {component, reduxThunkTester} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: false}});
    expect(reduxThunkTester.getActionHistoryStringify()).toMatchSnapshot();
  });

  test('UnmountField: render with field and then remove it (store)', () => {
    const {component, store} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: false}});
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });

  test('UnmountField: mount field, add error and then remove it (store)', () => {
    const {component, store} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    component.find('input').at(2).simulate('change', {target: {value: ''}});
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: false}});
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });

  test('UnmountField: mount field, add warning and then remove it (store)', () => {
    const {component, store} = renderComponent();
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: true}});
    component.find('input').at(2).simulate('change', {target: {value: 't'}});
    component.find('input').at(1).simulate('change', {target: {type: 'checkbox', checked: false}});
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });
});
