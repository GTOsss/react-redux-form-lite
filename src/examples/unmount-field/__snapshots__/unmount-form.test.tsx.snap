// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnmountField /> Render UnmountField 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ReduxForm)>
    <ReduxForm
      actions={
        Object {
          "arrayPush": [Function],
          "blur": [Function],
          "change": [Function],
          "changeSubmitted": [Function],
          "focus": [Function],
          "registerField": [Function],
          "registerForm": [Function],
          "removeField": [Function],
          "removeForm": [Function],
          "updateFormState": [Function],
          "updateValidateAndWarningMessages": [Function],
          "updateValidateMessage": [Function],
          "updateValidateMessages": [Function],
          "updateWarningMessage": [Function],
          "updateWarningMessages": [Function],
        }
      }
      formState={
        Object {
          "form": Object {
            "activeField": "",
            "blurred": false,
            "changed": false,
            "errorsMap": Object {},
            "focused": false,
            "hasErrors": false,
            "hasWarnings": false,
            "submitted": false,
            "warningsMap": Object {},
          },
          "meta": Object {
            "firstName": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "",
              "focused": false,
              "warning": "",
            },
            "withLastName": Object {
              "active": false,
              "blurred": false,
              "changed": false,
              "error": "",
              "focused": false,
              "warning": "",
            },
          },
          "values": Object {
            "firstName": undefined,
            "withLastName": undefined,
          },
        }
      }
      ownProps={Object {}}
    >
      <UnmountForm
        formActions={
          Object {
            "arrayPush": [Function],
            "blur": [Function],
            "change": [Function],
            "changeSubmitted": [Function],
            "focus": [Function],
            "registerField": [Function],
            "registerForm": [Function],
            "removeField": [Function],
            "removeForm": [Function],
            "updateFormState": [Function],
            "updateValidateAndWarningMessages": [Function],
            "updateValidateMessage": [Function],
            "updateValidateMessages": [Function],
            "updateWarningMessage": [Function],
            "updateWarningMessages": [Function],
          }
        }
        formParams={
          Object {
            "destroyOnUnmount": true,
            "form": "example",
          }
        }
        formState={
          Object {
            "form": Object {
              "activeField": "",
              "blurred": false,
              "changed": false,
              "errorsMap": Object {},
              "focused": false,
              "hasErrors": false,
              "hasWarnings": false,
              "submitted": false,
              "warningsMap": Object {},
            },
            "meta": Object {
              "firstName": Object {
                "active": false,
                "blurred": false,
                "changed": false,
                "error": "",
                "focused": false,
                "warning": "",
              },
              "withLastName": Object {
                "active": false,
                "blurred": false,
                "changed": false,
                "error": "",
                "focused": false,
                "warning": "",
              },
            },
            "values": Object {
              "firstName": undefined,
              "withLastName": undefined,
            },
          }
        }
        handleSubmit={[Function]}
      >
        <form
          onSubmit={[Function]}
        >
          <div>
            <label>
              First Name
            </label>
            <div>
              <FieldWithContext
                component="input"
                name="firstName"
                placeholder="First Name"
                type="text"
              >
                <Connect(Field)
                  component="input"
                  formContext={
                    Object {
                      "destroyOnUnmount": true,
                      "form": "example",
                      "updateValidateAndWarnMap": [Function],
                      "validate": [Function],
                      "warn": [Function],
                    }
                  }
                  name="firstName"
                  placeholder="First Name"
                  type="text"
                >
                  <Field
                    actions={
                      Object {
                        "arrayPush": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "changeSubmitted": [Function],
                        "focus": [Function],
                        "registerField": [Function],
                        "registerForm": [Function],
                        "removeField": [Function],
                        "removeForm": [Function],
                        "updateFormState": [Function],
                        "updateValidateAndWarningMessages": [Function],
                        "updateValidateMessage": [Function],
                        "updateValidateMessages": [Function],
                        "updateWarningMessage": [Function],
                        "updateWarningMessages": [Function],
                      }
                    }
                    component="input"
                    fieldArrayContext={Object {}}
                    formContext={
                      Object {
                        "destroyOnUnmount": true,
                        "form": "example",
                        "updateValidateAndWarnMap": [Function],
                        "validate": [Function],
                        "warn": [Function],
                      }
                    }
                    formState={
                      Object {
                        "form": Object {
                          "activeField": "",
                          "blurred": false,
                          "changed": false,
                          "errorsMap": Object {},
                          "focused": false,
                          "hasErrors": false,
                          "hasWarnings": false,
                          "submitted": false,
                          "warningsMap": Object {},
                        },
                        "meta": Object {
                          "firstName": Object {
                            "active": false,
                            "blurred": false,
                            "changed": false,
                            "error": "",
                            "focused": false,
                            "warning": "",
                          },
                          "withLastName": Object {
                            "active": false,
                            "blurred": false,
                            "changed": false,
                            "error": "",
                            "focused": false,
                            "warning": "",
                          },
                        },
                        "values": Object {
                          "firstName": undefined,
                          "withLastName": undefined,
                        },
                      }
                    }
                    name="firstName"
                    ownProps={
                      Object {
                        "name": "firstName",
                        "placeholder": "First Name",
                        "type": "text",
                      }
                    }
                  >
                    <input
                      name="firstName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="First Name"
                      type="text"
                      value=""
                    />
                  </Field>
                </Connect(Field)>
              </FieldWithContext>
            </div>
          </div>
          <FieldWithContext
            component="input"
            name="withLastName"
            type="checkbox"
          >
            <Connect(Field)
              component="input"
              formContext={
                Object {
                  "destroyOnUnmount": true,
                  "form": "example",
                  "updateValidateAndWarnMap": [Function],
                  "validate": [Function],
                  "warn": [Function],
                }
              }
              name="withLastName"
              type="checkbox"
            >
              <Field
                actions={
                  Object {
                    "arrayPush": [Function],
                    "blur": [Function],
                    "change": [Function],
                    "changeSubmitted": [Function],
                    "focus": [Function],
                    "registerField": [Function],
                    "registerForm": [Function],
                    "removeField": [Function],
                    "removeForm": [Function],
                    "updateFormState": [Function],
                    "updateValidateAndWarningMessages": [Function],
                    "updateValidateMessage": [Function],
                    "updateValidateMessages": [Function],
                    "updateWarningMessage": [Function],
                    "updateWarningMessages": [Function],
                  }
                }
                component="input"
                fieldArrayContext={Object {}}
                formContext={
                  Object {
                    "destroyOnUnmount": true,
                    "form": "example",
                    "updateValidateAndWarnMap": [Function],
                    "validate": [Function],
                    "warn": [Function],
                  }
                }
                formState={
                  Object {
                    "form": Object {
                      "activeField": "",
                      "blurred": false,
                      "changed": false,
                      "errorsMap": Object {},
                      "focused": false,
                      "hasErrors": false,
                      "hasWarnings": false,
                      "submitted": false,
                      "warningsMap": Object {},
                    },
                    "meta": Object {
                      "firstName": Object {
                        "active": false,
                        "blurred": false,
                        "changed": false,
                        "error": "",
                        "focused": false,
                        "warning": "",
                      },
                      "withLastName": Object {
                        "active": false,
                        "blurred": false,
                        "changed": false,
                        "error": "",
                        "focused": false,
                        "warning": "",
                      },
                    },
                    "values": Object {
                      "firstName": undefined,
                      "withLastName": undefined,
                    },
                  }
                }
                name="withLastName"
                ownProps={
                  Object {
                    "name": "withLastName",
                    "type": "checkbox",
                  }
                }
              >
                <input
                  name="withLastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                  value=""
                />
              </Field>
            </Connect(Field)>
          </FieldWithContext>
          <div>
            <button
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </UnmountForm>
    </ReduxForm>
  </Connect(ReduxForm)>
</Provider>
`;

exports[`<UnmountField /> Render UnmountField: inputs 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="withLastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value=""
  />,
]
`;

exports[`<UnmountField /> Render UnmountField: inputs after change 1`] = `
Array [
  <input
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="withLastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="checkbox"
    value={true}
  />,
  <input
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<UnmountField /> Render UnmountField: store 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "withLastName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": undefined,
    "withLastName": undefined,
  },
}
`;

exports[`<UnmountField /> UnmountField: mount field (action history) 1`] = `
"@@react-redux-form-lite/REGISTER_FORM
 meta: {form: 'example'},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {form: 'example', field: 'firstName'}
 payload: {value: undefined},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {form: 'example', field: 'withLastName'}
 payload: {value: undefined},
@@react-redux-form-lite/CHANGE
 meta: {form: 'example', field: 'withLastName'}
 payload: {value: true},
@@react-redux-form-lite/UPDATE_VALIDATE_AND_WARNING_MESSAGES
 meta: {form: 'example'}
 payload: {
	value: {
		errors: {lastName: 'Field required.'},
		warnings: {}
	},
	submitted: undefined
},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {form: 'example', field: 'lastName'}
 payload: {value: undefined}"
`;

exports[`<UnmountField /> UnmountField: mount field (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "lastName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "withLastName": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": undefined,
    "lastName": undefined,
    "withLastName": true,
  },
}
`;

exports[`<UnmountField /> UnmountField: mount field, add error and then remove it (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "withLastName": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": undefined,
    "withLastName": false,
  },
}
`;

exports[`<UnmountField /> UnmountField: mount field, add warning and then remove it (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "withLastName": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": undefined,
    "withLastName": false,
  },
}
`;

exports[`<UnmountField /> UnmountField: render with field and then remove it (action history) 1`] = `
"@@react-redux-form-lite/REGISTER_FORM
 meta: {form: 'example'},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {form: 'example', field: 'firstName'}
 payload: {value: undefined},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {form: 'example', field: 'withLastName'}
 payload: {value: undefined},
@@react-redux-form-lite/CHANGE
 meta: {form: 'example', field: 'withLastName'}
 payload: {value: true},
@@react-redux-form-lite/UPDATE_VALIDATE_AND_WARNING_MESSAGES
 meta: {form: 'example'}
 payload: {
	value: {
		errors: {lastName: 'Field required.'},
		warnings: {}
	},
	submitted: undefined
},
@@react-redux-form-lite/REGISTER_FIELD
 meta: {form: 'example', field: 'lastName'}
 payload: {value: undefined},
@@react-redux-form-lite/CHANGE
 meta: {form: 'example', field: 'withLastName'}
 payload: {value: false},
@@react-redux-form-lite/UPDATE_VALIDATE_AND_WARNING_MESSAGES
 meta: {form: 'example'}
 payload: {
	value: {
		errors: {lastName: 'Field required.'},
		warnings: {}
	},
	submitted: undefined
},
@@react-redux-form-lite/REMOVE_FIELD
 meta: {form: 'example', field: 'lastName'}"
`;

exports[`<UnmountField /> UnmountField: render with field and then remove it (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": true,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "firstName": Object {
      "active": false,
      "blurred": false,
      "changed": false,
      "error": "",
      "focused": false,
      "warning": "",
    },
    "withLastName": Object {
      "active": false,
      "blurred": false,
      "changed": true,
      "error": "",
      "focused": false,
      "warning": "",
    },
  },
  "values": Object {
    "firstName": undefined,
    "withLastName": false,
  },
}
`;
